ext/sodium/libsodium.c

```diff
@@ -3405,23 +3280,23 @@ PHP_FUNCTION(sodium_crypto_secretstream_xchacha20poly1305_push)
 	}
 	ZVAL_DEREF(state_zv);
 	if (Z_TYPE_P(state_zv) != IS_STRING) {
-		zend_throw_exception(sodium_exception_ce, "a reference to a state is required", 0);
+		zend_argument_error(sodium_exception_ce, 1, "to be a reference to a state");
 		RETURN_THROWS();
 	}
 	sodium_separate_string(state_zv);
 	state = (unsigned char *) Z_STRVAL(*state_zv);
 	state_len = Z_STRLEN(*state_zv);
 	if (state_len != sizeof (crypto_secretstream_xchacha20poly1305_state)) {
-		zend_throw_exception(sodium_exception_ce, "incorrect state length", 0);
+		zend_argument_error(sodium_exception_ce, 1, "to have a correct length");
 		RETURN_THROWS();
 	}
 	if (msg_len > crypto_secretstream_xchacha20poly1305_MESSAGEBYTES_MAX ||
 		msg_len > SIZE_MAX - crypto_secretstream_xchacha20poly1305_ABYTES) {
-		zend_throw_exception(sodium_exception_ce, "message cannot be larger than SODIUM_CRYPTO_SECRETSTREAM_XCHACHA20POLY1305_MESSAGEBYTES_MAX bytes", 0);
+		zend_argument_error(sodium_exception_ce, 2, "to be shorter than or equal to SODIUM_CRYPTO_SECRETSTREAM_XCHACHA20POLY1305_MESSAGEBYTES_MAX bytes");
```

Comment:
Especially for cases like this, I think that phrasing this in the `to be at most xxx bytes long` style is more readable. "shorter than or equal" is pretty odd.

Reply:
Fixed, it was very odd indeed. :)